### DISCUSSION TITLE
How to pass stdin?

### DISCUSSION THREAD

**Discussion by ByteDrummer** [mfussenegger/nvim-dap#430] [Q&A] [answered]
I've been trying to figure out how to debug a program that reads from stdin. Can I modify the adapter configuration to specify stdin or pass stdin from a different terminal?

**Comment by mfussenegger** [comment, marked as answer]
Depends on the debug adapter. Some support starting an application in a terminal and wait for the debugger to attach.
There is no standardized way. See also https://github.com/microsoft/debug-adapter-protocol/issues/216

**Reply by ByteDrummer** [reply]
Thanks. I was looking at the VSCode docs and I guess in my case with python all I needed was `console = "integratedTerminal"` in the adapter config.

**Reply by mfussenegger** [reply]
With debugpy one option is to add the following in the application:

```python
import debugpy
debugpy.listen(5678)
debugpy.wait_for_client()
```

Then start the application regularly in a terminal and attach the debugger to it.
